aws/resource_aws_s3_bucket.go

```diff
@@ -65,9 +65,53 @@ func resourceAwsS3Bucket() *schema.Resource {
 			},
 
 			"acl": {
-				Type:     schema.TypeString,
-				Default:  "private",
-				Optional: true,
+				Type:          schema.TypeString,
+				Default:       "private",
+				Optional:      true,
+				ConflictsWith: []string{"grant"},
+			},
```

Comment:
How does having a default value specified interact with `ConflictsWith`?

Either way we might want to consider dropping `acl` all together, since it's not something we scan from AWS (every bucket in a scan thinks it's `private` since that's the default). Otherwise there may be some changes we need  to make in the tf.json generation code to provide a preference for `grant` over `acl`.